tests/tester.nim

```diff
@@ -859,4 +860,40 @@ test "do not install single dependency multiple times (#678)":
       check execNimble(["refresh"]).exitCode == QuitSuccess
       let (output, exitCode) = execNimble("install", "-y")
       check exitCode == QuitSuccess
-      check output.find("issue678_dependency_1@0.1.0 already exists") == -1
+      let index = output.find("issue678_dependency_1@0.1.0 already exists")
+      check index == stringNotFound
+
+test "compilation without warnings":
```

Comment:
Not too sure I want this to be a test. Also, doesn't Nim offer a flag that turns warnings into errors?